src/components/data/data-utils.js

```diff
@@ -360,19 +352,29 @@ export const EntityUploader = ({ onSuccess, onDismiss, namespace, name, entityTy
         title: 'Import Table Data',
         width: '35rem',
         okButton: h(ButtonPrimary, {
-          disabled: !currentFile || isInvalid || uploading,
-          tooltip: !currentFile || isInvalid ? 'Please select valid data to upload' : 'Upload selected data',
+          disabled: dataProvider.tsvFeatures.disabled(currentFile, isInvalid, uploading, recordType),
+          tooltip: dataProvider.tsvFeatures.tooltip(currentFile, isInvalid, recordType),
           onClick: doUpload
         }, ['Start Import Job'])
       }, [
         div({ style: { padding: '0 0 1rem' } },
-          ['Choose the data import option below. ',
+          [dataProvider.tsvFeatures.uploadInstructions,
             h(Link, {
               ...Utils.newTabLinkProps,
               href: 'https://support.terra.bio/hc/en-us/articles/360025758392'
             }, ['Click here for more info on the table.']),
             p(['Data will be saved in location: 🇺🇸 ', span({ style: { fontWeight: 'bold' } }, 'US '), '(Terra-managed).'])]),
-        h(SimpleTabBar, {
+        dataProvider.tsvFeatures.needsTypeInput && div([div({ style: { padding: '0 0 1rem' } }, 'Table name:  '),
+          h(TextInput, {
```

Comment:
Could this input field validate that the user's input is a valid record type (can't start with sys_ or contain characters besides letters, numbers, spaces, dashes or underscores)? We do this in the input form for workspace name when creating a new workspace, for instance.

I'd be okay if this is a follow-on ticket after this PR merges.